Sources/SwiftSyntaxMacroExpansion/MacroReplacement.swift

```diff
@@ -254,31 +325,72 @@ private final class MacroExpansionRewriter: SyntaxRewriter {
     // Swap in the argument for this parameter
     return arguments[parameterIndex].trimmed
   }
+
+  override func visit(_ node: GenericArgumentSyntax) -> GenericArgumentSyntax {
+    guard let parameterIndex = genericParameterReplacements[node] else {
+      return super.visit(node)
+    }
+
+    guard parameterIndex < genericArguments.count else {
+      return super.visit(node)
+    }
+
+    // Swap in the argument for type parameter
+    return GenericArgumentSyntax(
+      leadingTrivia: node.leadingTrivia,
+      node.unexpectedBeforeArgument,
+      argument: genericArguments[parameterIndex].trimmed,
+      node.unexpectedBetweenArgumentAndTrailingComma,
+      trailingComma: node.trailingComma,
+      node.unexpectedAfterTrailingComma
+        // TODO: seems we're getting spurious trailing " " here,
+        //  skipping trailing trivia for now
+        // trailingTrivia: node.trailingTrivia
```

Comment:
Do you have a failing test case for this? I tried adding it again and none of the tests failed.

Reply:
Oh, weird... yes I would get a weird extra space after `Int, [ ]String` like this. Let me check again hmm